tests/unit/matrix/SparseTests.hpp

```diff
@@ -0,0 +1,346 @@
+#pragma once
+#include <string>
+#include <vector>
+#include <sstream>
+#include <iterator>
+#include <algorithm>
+#include <cassert>
+#include <tests/unit/TestBase.hpp>
+#include <resolve/vector/Vector.hpp>
+#include <resolve/matrix/Csr.hpp>
+#include <resolve/matrix/MatrixHandler.hpp>
+#include <resolve/MemoryUtils.hpp>
+
+namespace ReSolve { namespace tests {
+  class SparseTests : TestBase
+  {
+  public:
+    SparseTests(ReSolve::MatrixHandler& handler) : handler_(handler)
+    {
+      if (handler_.getIsCudaEnabled() || handler_.getIsHipEnabled()) {
+        memspace_ = memory::DEVICE;
+      } else {
+        memspace_ = memory::HOST;
+      }
+    }
+
+    ~SparseTests()
+    {
+    }
+
+    /**
+     * @brief Constructor test for sparse matrix
+     * 
+     * Constructs a CSR matrix with the given parameters and confirms that the
+     * parameters are stored correctly.
+     * 
+     * @param[in] n - number of rows
+     * @param[in] m - number of columns
+     * @param[in] nnz - number of non-zeros
+     * @param[in] is_symmetric - true if the matrix is symmetric, false otherwise
+     * @param[in] is_expanded - true if the matrix is expanded (all non-zeros 
+     are stored explicitly, not assumed based on symmetry), false otherwise
+     * 
+     * @return TestOutcome indicating success or failure of the test
+     */
+    TestOutcome constructor(index_type n, 
+                            index_type m, 
+                            index_type nnz,
+                            bool is_symmetric = false,
+                            bool is_expanded = true) 
+    {
+      TestStatus status;
+      status = true;
+
+      ReSolve::matrix::Csr A(n, m, nnz, is_symmetric, is_expanded);
+
+      if (A.getNumRows() != n || A.getNumColumns() != m || A.getNnz() != nnz) {
+        std::cout << "Matrix dimensions do not match expected values.\n";
+        status = false;
+      }
+
+      return status.report(__func__);
+    }
+
+    /**
+     * @brief Test set data pointers for the sparse matrix
+     * 
+     * Sets the row, column, and value data pointers for the sparse matrix and
+     * checks if the pointers are set correctly.
+     * 
+     * @param n - number of rows
+     * @param m - number of columns
+     * @param nnz - number of non-zeros
+     * @return TestOutcome indicating success or failure of the test
+     */
+    TestOutcome setDataPointers(index_type n, index_type m, index_type nnz)
+    {
+      assert(nnz % n == 0 && "For this test, nnz must be divisible by m");
+      
+      TestStatus status;
+      status = true;
```

Comment:
Consider "success" as the variable name rather than "status". This makes intent of the code more clear.
```suggestion
      TestStatus success;
      success = true;
```